aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -53,17 +53,26 @@ static inline Tensor unary_op_impl(const Tensor& self, OutImpl& out_impl) {
   return out_impl(result, self);
 }
 
+// Implicitly promote all int and bool dtypes to default floating dtype
```

Comment:
Let's put enough into this comment that we don't need to reference the PR and people know when to use this function instead of unary_op_impl. What are the key points?

1. This ensures the output type of the function is a floating point value.
2. It's appropriate for functions like sin() which (mathematically speaking) produce non-integral values.
3. Implementers should continue to use the other unary functions, like unary_op_impl_, since they explicitly specify the output type. 

Does that sound reasonable? Also, the promotion is not implicit (it's very explicit).

Reply:
Okay, sure! I'll add a detailed comment containing the above key points instead of linking to the PR.

Reply:
Added the comments! :) (https://github.com/pytorch/pytorch/pull/33322/commits/c816f39f45369536c2ffb329756ae36405b21d64)